### DISCUSSION TITLE
Support Other Puppet Rendering Systems

### DISCUSSION THREAD

**Discussion by sr229** [vignetteapp/vignette#215] [Ideas]
It has become apparent that we need to support multiple rendering systems in order to reach a wide audience. In order to make this possible, we need to make the puppet implementation abstract and make it modularized for end users to use whatever they see fit for their needs. The following listed below is what we'd want to support initially.

- **Live2D**
    - Apparently became a standard as used in workflows by many VTubers indie or company-wise.
- **Rive**
    - A free alternative software-wise. It's editor however isn't and requires to be paid.
- **Inochi2D**
    - A fully free alternative from its editor to its runtime. It is at a highly experimental stage however.

**Comment by LeNitrous** [comment]
Coming back to this, I will wait for Inochi2D so I can compare it against Live2D to see what differences and similarities both systems have so I can be able to make abstracted APIs I can start with as a foundation.
